pgjdbc/src/test/java/org/postgresql/test/jdbc2/ResultSetTest.java

```diff
@@ -643,10 +645,55 @@ public void testgetLong() throws SQLException {
       try {
         rs.getLong(1);
         fail("Exception expected." + rs.getString(1));
-      } catch (Exception e) {
+      } catch (SQLException e) {
       }
     }
     rs.close();
+
+    rs = con.createStatement().executeQuery("select a from testboolfloat");
```

Comment:
The query has no order by, so the order of rows can be different, so the test would have flaky failures.
 Would you please split the test in several ones?